### ISSUE TITLE
seek not working

### ISSUE THREAD

**Issue by porjo** [jessuni/shikwasa#75]
Thanks for the player!

I had an issue where calling seek on a newly loaded player wasn't working. I found I had to delay the seek to give the player enough time to load enough audio to get a duration value e.g.

```js
                       let ss = setInterval(() => {
				if(player.duration > 0) {
					clearInterval(ss);
					player.seek(lastPlayTimeSec);
				}
			},500);
```

I'm wondering if something like that should be built into the seek function itself?

**Comment by jessuni** [comment]
Hi. This is due to the audio not being ready for time-specific mutations, as it is so for any native `audio` element.
To check when the audio has loaded the necessary data, you can add a listener on `loadedmetadata`, that's when the audio is ready for further action. You can read further [here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/loadedmetadata_event).

```javascript
// initing the player

console.log(player.seekable) // false
console.log(player.audio.readyState) // 0, this is the native audio property that indicates whether audio is ready or not
player.on('loadedmetadata',() => {
  console.log(player.seekable) // true
  console.log(player.audio.readyState) // should be 1 or above
})
```

Hope the explanation helps : )

**Comment by porjo** [comment]
Thanks for the explanation. 

I'm wondering how best to help others avoid the mistake I made. Perhaps the API doco could be updated to include the `seekable` property with a note about it becoming `true` after `loadedmetadata` event fires.

**Comment by jessuni** [comment]
Indeed, I was thinking about the same thing, as not everyone using the library is familiar with the `HTMLMediaElement` API. Thanks!

**Comment by emmadesilva** [comment]
Also make sure your web server properly handles this! Running through the built in PHP webserver this (and normal scroll) did not work. Once on GitHub Pages, it works great!
